lib/ASTGen/Sources/ASTGen/PluginMessages.swift

```diff
@@ -49,11 +51,18 @@ internal enum PluginToHostMessage: Codable {
     capability: PluginMessage.PluginCapability
   )
 
+  case expandMacroResult(
+    expandedSource: String?,
+    diagnostics: [PluginMessage.Diagnostic]
+  )
```

Comment:
Another idea is, keep `expandFreestandingMacroResult` as-is, and make `expandAttachedMacroResult` be
```swift
  case expandAttachedMacroResult(
     expandedSource: String?,
     expandedSources: [String]?,
     diagnostics: [PluginMessage.Diagnostic]
  )
```
then plugins fill either `expandedSource` or `expandedSources` depending on host's protocol version.
@bnbarham wdyt?